wiring/src/spark_wiring_ble.cpp

```diff
@@ -2436,8 +2434,8 @@ class BleScanDelegator {
         size_t filterCustomDatalen;
         const uint8_t* filterCustomData = filter_.customData(&filterCustomDatalen);
         if (filterCustomData != nullptr && filterCustomDatalen > 0) {
-            size_t srLen = result.scanResponse().get(BleAdvertisingDataType::MANUFACTURER_SPECIFIC_DATA, nullptr, sizeof(size_t));
-            size_t advLen = result.advertisingData().get(BleAdvertisingDataType::MANUFACTURER_SPECIFIC_DATA, nullptr, sizeof(size_t));
+            size_t srLen = result.scanResponse().get(BleAdvertisingDataType::MANUFACTURER_SPECIFIC_DATA, nullptr, filterCustomDatalen);
```

Comment:
We should not try to get length of the custom data from the actual adv data. Think of this scenario:

- The custom data we're going to filter is `01 02 03 04`
- The actual custom data is the adv data is `01 02 03 04 aa bb cc`

Can we say that they are identical if we only compare the first `filterCustomDatalen` of data?

Reply:
Good point! I will set the length maximum to the maximum for each one.